### PR TITLE
feat: update/add pageSeparator params to LlamaParseReader

### PR DESCRIPTION
**Pull request by KindOfAScam**

### CODE REVIEW

File: apps/docs/docs/modules/data_loaders/discord.mdx [hunk 1]
```diff
@@ -20,7 +20,7 @@ Copy the URL in your browser and select the server you want your bot to join.
 #### DiscordReader()
 
 - `discordToken?`: The Discord bot token.
-- `makeRequest?`: Optionally provide a custom request function for edge environments, e.g. `fetch`. See discord.js for more info.
+- `requestHandler?`: Optionally provide a custom request function for edge environments, e.g. `fetch`. See discord.js for more info.
```
Comment (marcusschiesser) [inline apps/docs/docs/modules/data_loaders/discord.mdx:23]: why this change in the llamaparse PR?

File: apps/docs/docs/modules/data_loaders/llama_parse/index.mdx [hunk 2]
```diff
@@ -41,7 +41,9 @@ They can be divided into two groups.
 - `doNotCache?` Optional. Set to true to not cache the document.
 - `fastMode?` Optional. Set to true to use the fast mode. This mode will skip OCR of images, and table/heading reconstruction. Note: Non-compatible with `gpt4oMode`.
 - `doNotUnrollColumns?` Optional. Set to true to keep the text according to document layout. Reduce reconstruction accuracy, and LLMs/embeddings performances in most cases.
-- `pageSeparator?` Optional. The page separator to use. Defaults is `\\n---\\n`.
+- `pageSeparator?` Optional. A templated page separator to use to split the text. If it contain `{page_number}`, it will be replaced by the next page number. If not set the default separator `\\n---\\n` will be used.
```
Comment (marcusschiesser) [inline apps/docs/docs/modules/data_loaders/llama_parse/index.mdx:44]: Why does the `pageSeparator` contain the `{page_number}`?

File: apps/docs/docs/modules/data_loaders/llama_parse/index.mdx [hunk 3]
```diff
@@ -41,7 +41,9 @@ They can be divided into two groups.
...
+- `pagePrefix?` Optional. A templated prefix to add to the beginning of each page. If it contain `{page_number}`, it will be replaced by the page number.
+- `pageSuffix?` Optional. A templated suffix to add to the end of each page. If it contain `{page_number}`, it will be replaced by the page number.
```
Comment (marcusschiesser) [inline apps/docs/docs/modules/data_loaders/llama_parse/index.mdx:46]:
```suggestion
- `pagePrefix?` Optional. A templated prefix to add to the beginning of each page. The variable `{page_number}` will be replaced by the page number.
- `pageSuffix?` Optional. A templated suffix to add to the end of each page. The variable `{page_number}` will be replaced by the page number.
```

File: packages/llamaindex/src/readers/LlamaParseReader.ts [hunk 4]
```diff
@@ -127,8 +127,12 @@ export class LlamaParseReader extends FileReader {
   fastMode?: boolean;
   // Wether to keep column in the text according to document layout. Reduce reconstruction accuracy, and LLM's/embedings performances in most cases.
   doNotUnrollColumns?: boolean;
-  // The page separator to use to split the text. Default is None, which means the parser will use the default separator '\\n---\\n'.
+  // A templated page separator to use to split the text. If it contain `{page_number}`, it will be replaced by the next page number. If not set the default separator '\\n---\\n' will be used.
   pageSeparator?: string;
+  //A templated prefix to add to the beginning of each page. If it contain `{page_number}`, it will be replaced by the page number.>
+  pagePrefix?: string;
+  // A templated suffix to add to the end of each page. If it contain `{page_number}`, it will be replaced by the page number.
+  pageSuffix?: string;
```
Comment (marcusschiesser) [inline packages/llamaindex/src/readers/LlamaParseReader.ts:135]: sync with .md comments

File: packages/llamaindex/src/readers/LlamaParseReader.ts [hunk 5]
```diff
@@ -207,6 +213,15 @@ export class LlamaParseReader extends FileReader {
       vendor_multimodal_api_key: this.vendorMultimodalApiKey,
     };
 
+    // Filter out invalid specific parameters
+    const filteredParams = this.filterSpecificParams(LlamaParseBodyParams, [
```
Comment (marcusschiesser) [inline packages/llamaindex/src/readers/LlamaParseReader.ts:217]: where is this used?

File: apps/docs/docs/modules/data_loaders/discord.mdx (hunk 1 above)
Comment (KindOfAScam) [reply apps/docs/docs/modules/data_loaders/discord.mdx:23]: I just noticed it and it's just a change in the docs, I mistakenly called it `makeRequest` in my previous PR, which is not the actual param name.

File: apps/docs/docs/modules/data_loaders/llama_parse/index.mdx (hunk 2 above)
Comment (KindOfAScam) [reply apps/docs/docs/modules/data_loaders/llama_parse/index.mdx]: Sorry, I just copied the python description. 
I assume it's currently only relevant for JSON mode where a page_number is returned.
Improved the descriptions.

File: packages/llamaindex/src/readers/LlamaParseReader.ts (hunk 5 above)
Comment (KindOfAScam) [reply packages/llamaindex/src/readers/LlamaParseReader.ts:217]: Oh yeah.. I did forget to actually use it.

File: apps/docs/docs/modules/data_loaders/discord.mdx (hunk 1 above)
Comment (marcusschiesser) [reply apps/docs/docs/modules/data_loaders/discord.mdx:23]: @KindOfAScam Great catch! Thanks a lot. To avoid confusion, better send a separate PR next time

**Review by marcusschiesser — APPROVED** [review]
Great job @KindOfAScam
